hipe4ml/model_handler.py

```diff
@@ -373,8 +381,8 @@ def cast_model_params(self, params):
             Hyperparameter values updated
         """
         for key in params.keys():
-            if isinstance(self.model.get_params()[key], int):
-                params[key] = int(round(params[key]))
+            if key in self.model.get_params():
+                params[key] = type(self.model.get_params()[key])(params[key])
         return params
 
```

Comment:
Now the usage of `cast_model_params` is a bit tricky. I would specify that the model parameters have to be correctly initialized before calling this function and I would also make it private.

Reply:
Agreed, a put a comment in the docstring and I made the function private